sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -102,73 +135,258 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
         .innerJoin("offering.institutionLocation", "institutionLocation")
         .getOne();
 
-      // Cloning existing offering.
-      const offering: EducationProgramOffering = { ...existingOffering };
-
-      // Assigning id as undefined, so that when its saved its considered as a new EducationProgramOffering object.
-      offering.id = undefined;
-
-      const newStudyEndDate =
-        scholasticStandingData.dateOfChange ??
-        scholasticStandingData.dateOfCompletion ??
-        scholasticStandingData.dateOfIncompletion ??
-        scholasticStandingData.dateOfWithdrawal;
-
-      offering.studyEndDate = new Date(newStudyEndDate);
-
-      offering.actualTuitionCosts =
-        scholasticStandingData.tuition ?? existingOffering.actualTuitionCosts;
-      offering.programRelatedCosts =
-        scholasticStandingData.booksAndSupplies ??
-        existingOffering.programRelatedCosts;
-      offering.mandatoryFees =
-        scholasticStandingData.mandatoryFees ?? existingOffering.mandatoryFees;
-      offering.exceptionalExpenses =
-        scholasticStandingData.exceptionalCosts ??
-        existingOffering.exceptionalExpenses;
-
-      offering.offeringType = OfferingTypes.ScholasticStanding;
+      // Check for restrictions and apply if any.
+      const studentRestriction = await this.getStudentRestrictions(
```

Comment:
I believe that `SchilasticStanding` should be part of this method name, it is too specific and related to scholastic standing. I would suggest `getScholasticStandingRestrictions`